waku/v2/protocol/common_service.go

```diff
@@ -4,19 +4,21 @@ import (
 	"context"
 	"errors"
 	"sync"
+	"sync/atomic"
 )
 
 // this is common layout for all the services that require mutex protection and a guarantee that all running goroutines will be finished before stop finishes execution. This guarantee comes from waitGroup all one has to use CommonService.WaitGroup() in the goroutines that should finish by the end of stop function.
-type CommonService struct {
+type CommonService[T any] struct {
 	sync.RWMutex
 	cancel  context.CancelFunc
 	ctx     context.Context
 	wg      sync.WaitGroup
-	started bool
+	started atomic.Bool
```

Comment:
Did not get the need for an atomic here since this var is already protected by a mutex.